sdk/src/main/java/com/oursky/authgear/GrantType.kt

```diff
@@ -0,0 +1,42 @@
+package com.oursky.authgear
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+@Serializable(with = GrantTypeSerializer :: class)
+enum class GrantType(val raw: String) {
+    AUTHORIZATION_CODE("authorization_code"),
```

Comment:
Can you verify how does this enum look like in Java? If it is not compatible with Java, then we may just make the literal strings named constants.

Reply:
IIRC this is purely internal implementation detail so the enum class could be marked as internal, and as such, perhaps we don't need to care about java interop? Any chance we would have to expose this to users?